tests/python_package_test/test_sklearn.py

```diff
@@ -227,25 +231,107 @@ def test_grid_search(self):
+                      'eval_set': [(X_test, y_test)],
+                      'eval_metric': constant_metric,
+                      'early_stopping_rounds': 2}
+        rand = RandomizedSearchCV(lgb.LGBMClassifier(**params), param_dist, cv=2)
+        rand.fit(X_train, y_train, **fit_params)
+        score = rand.score(X_test, y_test)  # utilizes RandomizedSearchCV default refit=True
+        self.assertIn(rand.best_params_['boosting_type'], ['rf', 'gbdt'])
+        self.assertIn(rand.best_params_['n_estimators'], list(range(3, 10)))
+        self.assertGreaterEqual(rand.best_params_['reg_alpha'], 0.01)  # Left-closed boundary point
+        self.assertLessEqual(rand.best_params_['reg_alpha'], 0.06)  # Right-closed boundary point
+        self.assertLess(rand.best_score_, 1.0)
+        self.assertLess(rand.best_estimator_.best_score_['valid_0']['multi_logloss'], 0.25)
+        self.assertEqual(rand.best_estimator_.best_score_['valid_0']['error'], 0)
+        self.assertGreaterEqual(score, 0.2)
+        self.assertLessEqual(score, 1.)
+
+    def test_multioutput_classifier(self):
+        X, y = make_multilabel_classification(n_classes=3, random_state=0)
+        y = y.astype(str)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+        clf = MultiOutputClassifier(estimator=lgb.LGBMClassifier(n_estimators=10))
+        clf.fit(X_train, y_train)
+        score = clf.score(X_test, y_test)
+        self.assertGreaterEqual(score, 0.2)
+        self.assertLessEqual(score, 1.)
+        for classifier in clf.estimators_:
+            self.assertIsInstance(classifier.booster_, lgb.Booster)
+
+    # sklearn < 0.23 does not have as_frame parameter
+    @unittest.skipIf(sk_version < '0.23.0', 'scikit-learn version is less than 0.23')
+    def test_multioutput_regressor(self):
+        bunch = load_linnerud(as_frame=True)  # returns a Bunch instance
+        X, y = bunch['data'], bunch['target']
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+        reg = MultiOutputRegressor(estimator=lgb.LGBMRegressor(n_estimators=10))
+        reg.fit(X_train, y_train)
+        y_pred = reg.predict(X_test)
+        _, score, _ = mse(y_test, y_pred)
+        self.assertGreaterEqual(score, 0.2)
+        self.assertLessEqual(score, 120.0)
+        for regressor in reg.estimators_:
+            self.assertIsInstance(regressor.booster_, lgb.Booster)
+
+    def test_classifier_chain(self):
+        X, y = make_multilabel_classification(n_classes=3, random_state=0)
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+        clf = ClassifierChain(base_estimator=lgb.LGBMClassifier(n_estimators=10), order=[2, 0, 1])
+        clf.fit(X_train, y_train)
+        score = clf.score(X_test, y_test)
+        self.assertGreaterEqual(score, 0.2)
+        self.assertLessEqual(score, 1.)
+        for classifier in clf.estimators_:
```

Comment:
Maybe we can assert length of `classes_` and `order_` attributes as well after fitting?